### PR TITLE
Minor updates for channel

### PR DESCRIPTION
**Pull request by reyamir**
**Add**

- Add db_only to `listChannels`
- Add `getMeta` function
- Add `setMeta` function
- Update test for channel

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -4,8 +4,7 @@ import { Filter } from 'nostr-tools';
 import { NostrPool, NostrEvent } from '.';
 
 export async function listChannels(pool: NostrPool): Promise<ChannelInfo[]> {
-  // todo: this should only use the store, not go and re-query stuff, being lazy to get things done
-  return (await pool.list([{ kinds: [40] }])).map((ent) => {
+  return (await pool.list([{ kinds: [40] }], true)).map((ent) => {
```
Comment (earonesty) [inline src/channel.ts:7]: db_only should be a parameter passed to listChannels i think, the same way the rest are.   at least until we're sure the db is "good".  its nice to be able to turn it off and check things.

File: src/channel.ts [hunk 2]
```diff
@@ -59,12 +58,27 @@ class Nip28Channel {
     return ev;
   }
 
-  async setMeta(meta: ChannelInfo) {
-    throw new Error('not implemented yet');
+  async setMeta(channel_id: string, meta: ChannelInfo) {
+    if (!channel_id) throw new Error('channel id is required');
```
Comment (earonesty) [inline src/channel.ts:62]: please add a test case

File: src/channel.ts (hunk 1 above)
Comment (reyamir) [reply src/channel.ts]: I've updated it

**Review by earonesty — APPROVED** [review]

**Comment by earonesty** [comment]
approved and merged.

I'd like to know what you think of the encrypted group chat stuff

If you want to take a glance at it

**Pull request merged**